matc glsl transpile version issue

Hi,

When `matc` optimizations are enabled, it seems to set `CodeGenerator::mTargetLanguage` to `SPIRV`, in which case shaders become `#version 450`:
https://github.com/google/filament/blob/251abf0545b9b4afcc49a21de5620e79a743ddad/libs/filamat/src/shaders/CodeGenerator.cpp#L61-L64

The problem is when transpiling back to GLSL (because optimizations are enabled), it configures `CompilerGLSL `with `GL_CORE_41`:
https://github.com/google/filament/blob/251abf0545b9b4afcc49a21de5620e79a743ddad/libs/filamat/src/GLSLPostProcessor.cpp#L350

This means it will output a `#version 410` shader and the problem with that is:

```
"0(158) : error C7532: global function unpackHalf2x16 requires "#version 420" or later
0(158) : error C0000: ... or #extension GL_ARB_shading_language_packing : enable
"
1:   #version 410
2:   struct BoneData
3:   {
4:   mat3x4 transform;
5:   uvec4 cof;
6:   };
7:   layout(std140) uniform FrameUniforms
8:   {
9:   mat4 a;
10:   mat4 b;
11:   mat4 c;
12:   mat4 d;
13:   mat4 e;
14:   mat4 f;
15:   mat4 g[4];
16:   vec4 h;
17:   vec4 i;
18:   vec3 j;
19:   float k;
...
```

--- a/libs/filamat/src/GLSLPostProcessor.cpp
+++ b/libs/filamat/src/GLSLPostProcessor.cpp
@@ -357,7 +357,7 @@ void GLSLPostProcessor::fullOptimization(const TShader& tShader,
         CompilerGLSL glslCompiler(move(spirv));
         glslCompiler.set_common_options(glslOptions);
 
-        if (tShader.getStage() == EShLangFragment && !glslOptions.es) {
+        if (!glslOptions.es) {
             // enable GL_ARB_shading_language_packing if available
             glslCompiler.add_header_line("#extension GL_ARB_shading_language_packing : enable");
         }

